dmscripts/notify_suppliers_with_incomplete_applications.py

```diff
@@ -0,0 +1,61 @@
+from dmapiclient import DataAPIClient
+from dmscripts.helpers import logging_helpers
+from dmscripts.helpers.auth_helpers import get_auth_token
+from dmscripts.helpers.email_helpers import scripts_notify_client
+from dmscripts.helpers.logging_helpers import logging
+from dmutils.email.exceptions import EmailError
+from dmutils.email.helpers import hash_string
+from dmutils.env_helpers import get_api_endpoint_from_stage
+
+
+NOTIFY_TEMPLATE_ID = '25c7c763-fe51-418f-8e51-130c391edc35'
+
+
+def send_notification(mail_client, message, email, supplier_id, dry_run):
+    prefix = "[Dry Run] " if dry_run else ""
+    mail_client.logger.info(
+        f"{prefix}Sending email to supplier '{supplier_id}' "
+        f"user '{hash_string(email)}'"
+    )
+    if not dry_run:
+        try:
+            mail_client.send_email(
+                email, NOTIFY_TEMPLATE_ID, {'message': message}, allow_resend=False
+            )
+        except EmailError as e:
+            mail_client.logger.error(
+                f"Error sending email to supplier '{supplier_id}' "
+                f"user '{hash_string(email)}': {e}"
+            )
+            return 1
+    return 0
+
+
+def notify_suppliers_with_incomplete_applications(framework, stage, notify_api_key, dry_run):
+    logger = logging_helpers.configure_logger({"dmapiclient": logging.INFO})
+    mail_client = scripts_notify_client(notify_api_key, logger=logger)
+    data_api_client = DataAPIClient(
+        base_url=get_api_endpoint_from_stage(stage), auth_token=get_auth_token('api', stage)
+    )
+    error_count = 0
+
+    for sf in data_api_client.find_framework_suppliers_iter(framework):
+        message = ''
+        if not sf.get('applicationCompanyDetailsConfirmed', None):
+            message += '* confirm your company details\n'
+        if not sf.get('declaration', {'status': None}).get('status', None) == 'complete':
+            message += '* finish your supplier declaration\n'
+        if not data_api_client.find_draft_services(sf['supplierId'], framework=framework).get('services', []):
```

Comment:
We might need to be a bit stricter on this clause. We want to check if the supplier has at least 1 draft service with status `submitted` - if they don't, we should include the line below in the email.